pkg/cluster/controller/router.go

```
@@ -237,7 +257,14 @@ func (r *ClusterRouter) sendRequest(addr, command string) (string, error) {
 }
 
 func (r *ClusterRouter) sendDataRequest(addr string, data []byte) (string, error) {
-	conn, err := net.DialTimeout("tcp", addr, r.timeout)
+	dialer := &net.Dialer{Timeout: r.timeout}
+	var conn net.Conn
+	var err error
+	if r.internalTLS != nil {
+		conn, err = tls.DialWithDialer(dialer, "tcp", addr, r.internalTLS)
```

golangci-lint (2.12.2)

[error] crypto/tls.DialWithDialer must not be called. use (*crypto/tls.Dialer).DialContext with NetDialer

(noctx)